### ISSUE TITLE
nvidia/GPU support?

### ISSUE THREAD

**Issue by mwilliammyers** [cjbassi/ytop#28]
What do you think about adding GPU support via one of [these](https://crates.io/search?q=nvml&sort=downloads) crates?

Have you looked into it?

I would be happy to open a PR if it is something we would like to add...

**Comment by cjbassi** [comment]
Yah that sounds good.

**Comment by mwilliammyers** [comment]
I think I should probably add GPU support to `psutil`?

**Comment by cjbassi** [comment]
That's a good question, maybe we could start by just implementing things in ytop and then see about upstreaming it if it makes sense since I'm not totally sure what exactly were implementing yet. Also, maybe check to see if theres a GPU rust crate already.

**Comment by mwilliammyers** [comment]
Cool.

As far as I can tell there are only bindings to the [Nvidia Management Library](https://developer.nvidia.com/nvidia-management-library-nvml) via one of [these](https://crates.io/search?q=nvml&sort=downloads) crates. I was thinking [nvml-wrapper](https://crates.io/crates/nvml-wrapper) looked like the best option. 

That being said, I would love to have cross platform & cross GPU manufacturer support if possible, so maybe I will keep looking...?

**Comment by cjbassi** [comment]
Yeah cross platform GPU support would be nice. maybe we could start with Nvidia and add that to ytop if you want.

**Comment by mwilliammyers** [comment]
Cool, I will try to open a PR ASAP.

**Comment by cjbassi** [comment]
Looks like i3status-rust does some Nvidia stuff: https://github.com/greshake/i3status-rust/blob/master/src/blocks/nvidia_gpu.rs

They could probably benefit from an Nvidia library too so that they could clean up their code a bit

**Comment by Seirdy** [comment]
[Radeontop](https://github.com/clbr/radeontop) (written in C) might also be a good point of reference for AMD cards.
